osbuild/formats_common.py

```diff
@@ -0,0 +1,45 @@
+"""
+Common procedures and classes between the v1 and v2 formats
+"""
+
+from abc import abstractmethod
+
+
+class OSbuildError(Exception):
+    """
+    All exceptions that want to be formated in json for a user program must overload OSbuildError.
+    They also must be declared in the map 'map_error_id'.
+    """
+
+    @abstractmethod
+    def payload_as_dict(self):
+        pass
+
+    @property
+    def ID(self):
+        return OSbuildError.map_error_id.get(self.__class__, 0)
+
+    map_error_id = {}
+
+    @staticmethod
+    def add_error_id(error_klass):
+        OSbuildError.map_error_id[error_klass.__class__] = len(OSbuildError.map_error_id)+1
```

Comment:
Does this work?  The ID map is attached to the class, but are we guaranteed to only have one persistent instance of the map?

Reply:
Well it depends on how it is accessed. As long as it is accessed via ClassName.map_name we fine, but if it's called via instance.map_name then there will be an issue.

Reply:
Perhaps if we need this map, it should be a private module member with methods to return it and make sure we only have one instance.  A class member isn't guaranteed to be unique.  There's no real, explicit way to do singletons in Python, but a module can't be instantiated twice, so we get that kind of guarantee at the module level.

Another issue I just realised is that this method uses the argument's `__class__` as the key and we pass a class as the argument, so the key will always be `type`.

Reply:
I changed that too in the end.

Reply:
Oh I haven't seen your comment before answering (I did not reload the page until now), well I changed everything int there to get rid of the map. But your idea can work too. Let's discuss again after I push my new version :)

Reply:
The map is gone in the new version.